test/openshift-test.js

```diff
@@ -0,0 +1,68 @@
+const test = require('tape');
+const OpenshiftTestAssistant = require('openshift-test-assistant');
+const openshiftAssistant = new OpenshiftTestAssistant();
+const path = require('path');
+const request = require('supertest');
+
+test('setup', (t) => {
+  openshiftAssistant.deploy({
+    'projectLocation': path.join(__dirname, '/..'),
+    'strictSSL': false
+  }).then(() => {
+    t.ok(true); // indicate success
+    t.end();
+  }).catch(reason => {
+    t.fail(reason);
+    t.end();
+  });
+});
+
+test('test openshift greeting with no query param', (t) => {
+  if (!openshiftAssistant.isReady()) {
```

Comment:
It might integrate better with `Promise`-based tests if your test assistant was wrapped in a `Promise`. E.g. 

```js
openshiftInstance.ready().then( asst => {
  request(...)
    .get('/api/greeting')
    // etc
```
Where `openshiftInstance` is defined as something like this...

```js
const openshiftInstance = {
  ready() {
    if (openshiftAssistant.isReady()) return Promise.resolve(openshiftAssistant);
    return new Promise((resolve, reject) => {
      // use setTimeout and friends to periodically check isReady() and resolve when it returns true
      // perhaps reject after some configurable number of attempts...
    });
```
A thought anyway...

Reply:
Well... I just looked at the code for `openshift-test-assistant` and it seems you are already doing essentially this with the `waitForReady()` function. So in that case, I would just do 

```js
openshiftAssistant.waitForReady()
  .then(_ => {
    request(...);
    // etc.
  });
```

Reply:
Well the meaning of the condition 
<code>
 if (!openshiftAssistant.isReady()) {
</code>
is not to wait for apllication to become ready. It is a workaround that should do something like "Run this test only if setup was successfull".
Maybe the name "isReady" is confusing. I will consider renaming it.